src/infrastructure/transaction/CreateTransactionFromDTO.ts

```diff
@@ -200,6 +200,20 @@ const CreateStandaloneTransactionFromDTO = (transactionDTO, transactionInfo, isE
                 : undefined,
             transactionInfo,
         ).setPayloadSize(transactionDTO.size);
+    } else if (type === TransactionType.MOSAIC_SUPPLY_REVOCATION) {
+        return new MosaicSupplyRevocationTransaction(
+            transactionDTO.network,
+            version,
+            deadline,
+            maxFee,
+            extractRecipient(transactionDTO.sourceAddress),
+            new Mosaic(new MosaicId(transactionDTO.mosaicId), UInt64.fromNumericString(transactionDTO.amount)),
```

Comment:
this method started with fewer transadtions, but since now would it be better to use switch you think?

Reply:
converted to swich